core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -131,18 +131,26 @@ object TiBatchWrite {
     // pending: https://internal.pingcap.net/jira/browse/TIDB-1628
 
     // TODO: if this write is update, TiDB reuses row_id. We need adopt this behavior.
-    val rowIDAllocator = RowIDAllocator.create(
+
+    // when primary is handle, it does not require allocate ids for each row.
+    val offset = if(tiTableInfo.isPkHandle) {
```

Comment:
if condition should be (should require allocate ids for each row):
1. without primary key or
2. with primary key && auto increase && rdd does not contains primary key column